src/prpy/planning/snap.py

```diff
@@ -128,7 +130,9 @@ def _Snap(self, robot, goal, **kw_args):
                                             norm_order=2,
                                             sampling_func=vdc)
 
-        with self.robot_checker_factory(robot) as robot_checker:
+        with self.robot_checker_factory(robot) as robot_checker, \
+        robot.CreateRobotStateSaver(Robot.SaveParameters.ActiveDOF | 
+                                Robot.SaveParameters.LinkTransformation):
```

Comment:
I don't think we need `ActiveDOF` here, since we're not changing `robot`'s active DOF indices.